### PR TITLE
Fixed loosing POST packets on web socket connection failure

### PR DESCRIPTION
**Pull request by OneSman7**
Fixed loosing POST packets on web socket connection failure and engine memory leak on server errors.

### CODE REVIEW

File: Source/SocketIO/Engine/SocketEngine.swift [hunk 1]
```diff
@@ -371,6 +371,9 @@ open class SocketEngine : NSObject, URLSessionDelegate, SocketEnginePollable, So
         fastUpgrade = false
         probing = false
         flushProbeWait()
+        if postWait.count != 0 {
```
Comment (nuclearace) [inline Source/SocketIO/Engine/SocketEngine.swift]: How about a `!.isEmpty`?

Comment (OneSman7) [reply Source/SocketIO/Engine/SocketEngine.swift]: No problem, I just copied existing code.

Comment (nuclearace) [reply Source/SocketIO/Engine/SocketEngine.swift]: Or maybe move that check into flush... and unconditionally call it?

Comment (nuclearace) [reply Source/SocketIO/Engine/SocketEngine.swift]: I'm fine with this way in that case! Would you mind also adding some documentation comments to these methods that explain this failure case? I think those will be helpful when I come back in 6 months and try to refactor and end up breaking it again :).

Comment (OneSman7) [reply Source/SocketIO/Engine/SocketEngine.swift]: 👌 will also change to `!.isEmpty` and test

**Pull request merged**